scipy/optimize/_differentialevolution.py

```diff
@@ -1030,16 +1066,21 @@ def solve(self):
                 break
 
             if self.disp:
-                print("differential_evolution step %d: f(x)= %g"
-                      % (nit,
-                         self.population_energies[0]))
+                print(f"differential_evolution step {nit}: f(x)="
+                      f" {self.population_energies[0]}"
+                      )
 
             if self.callback:
                 c = self.tol / (self.convergence + _MACHEPS)
-                warning_flag = bool(self.callback(self.x, convergence=c))
+                res = self._result(nit=nit)
```

Comment:
I'd suggest creating an "in progress" message. It also seems safe to pass `warning_flag=False` here since it hasn't been raised yet. Alternatively, these could be default values of `message` and `warning_flag` defined in `_result` rather than accepting `kwds`.

Reply:
The `intermediate_result.message = 'in progress'` has been added. The message will change when the solve finishes. `intermediate_result.success` will be `True` during minimisation.